test/bricks-net-api.test.js

```diff
@@ -0,0 +1,634 @@
+				_DEBUG_LOG('TESTS: Got request:', request.method, request.url, request.httpVersion, '\n', request.headers);
+				serverRequestHandler(request, response);
+			});
+			server.on('listening', function () {
+				_DEBUG_LOG('TESTS: Server is listening.');
+				done();
+			});
+			server.listen(20000);
+		});
+		afterEach(function (done) {
+			_DEBUG_LOG('TESTS: Server is shutting down...');
+			server.close(function () {
+				_DEBUG_LOG('TESTS: Server has been shut down.');
+				done();
+			});
+			while (sockets.length > 0) {
+				sockets.shift().destroy();
+			}
+		});
+		
+		it('should return a promise-like `HTTPResponse` that resolves to itself and has a `body`', function (done) {
+			var response = api.HTTP(api.GET("localhost:20000/test"));
+			
+			assert.equal(true, response instanceof api.HTTPResponse, 'response instanceof HTTPResponse');
+			assert.equal(true, when.isPromiseLike(response), 'response isPromiseLike');
+			
+			when(
+				response
+			).then(function (response) {
+				assert.equal(true, response instanceof api.HTTPResponse, 'promised response instanceof HTTPResponse');
+				assert.equal('string', typeof response.body);
+				assert.equal('OK', response.body);
+				done();
+			}).done(undefined, done);
+		});
+		
+		it('should make a `GET` request, response `body` is a promise-like that resolves to the response body', function (done) {
+			serverRequestHandler = function (request, response) {
+				try {
+					assert.equal('GET', request.method);
+					assert.equal('/test', request.url);
+					assert.equal('1.1', request.httpVersion);
+					
+					// WARNING: `request.rawHeaders` requires Node.js 0.12.0
+					assert.deepEqual([
+						'Host',
+						'localhost'
+					], request.rawHeaders);
+					
+					response.end('OK');
+				}
+				catch (ex) {
+					done(ex);
+				}
+			};
+			
+			var response = api.HTTP(api.GET("localhost:20000/test"));
+			
+			assert.equal(true, response instanceof api.HTTPResponse, 'response instanceof HTTPResponse');
+			assert.equal(true, when.isPromiseLike(response.body), 'response.body isPromiseLike');
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body);
+				assert.equal('OK', body);
+				done();
+			}).done(undefined, done);
+		});
+		
+		it('should handle response Content-Length', function (done) {
+			serverRequestHandler = function (request, response) {
+				response.writeHead(200, {
+					'Content-Length': 5
+				});
+				response.write('OK');
+				response.write('OK2');
+				response.end('Overflow');
+			};
+			
+			var response = api.HTTP(api.GET("localhost:20000/test"));
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body);
+				assert.equal('OKOK2', body);
+				done();
+			}).done(undefined, done);
+		});
+		
+		/* TODO(sompylasar): Investigate further on this test failure.
+		it('should throw ConnectionResetByPeer on early close', function (done) {
```

Comment:
Looks like there is [a concept of tagging and using --grep option](https://github.com/mochajs/mocha/wiki/Tagging). Should use it instead.